### ISSUE TITLE
Multiple gpfs mounts from dynamic provisioning not reflected in df output

### ISSUE THREAD

**Issue by przem123** [IBM/ibm-spectrum-scale-csi#81]
On Openshift 4.3 Webconsole I created mongoDB deployment from Developer Catalog with dynamically provisioned 1GB PVC (named `mongodb`) mounted at `/var/lib/mongodb/data`. After that, I selected the option 'Add storage' and created a new PVC (named `newdata1`) with mount point `/newdata1`. 

```
[root@worker01 ~]# oc describe pod mongodb-2-plpgl
Name:         mongodb-2-plpgl
Namespace:    przemek
Priority:     0
Node:         worker02.ocp4.scale.com/192.168.1.16
Start Time:   Fri, 14 Feb 2020 17:20:38 +0100
Labels:       deployment=mongodb-2
              deploymentconfig=mongodb
              name=mongodb
Annotations:  k8s.v1.cni.cncf.io/networks-status:
                [{
                    "name": "openshift-sdn",
                    "interface": "eth0",
                    "ips": [
                        "10.128.2.168"
                    ],
                    "dns": {},
                    "default-route": [
                        "10.128.2.1"
                    ]
                }]
              openshift.io/deployment-config.latest-version: 2
              openshift.io/deployment-config.name: mongodb
              openshift.io/deployment.name: mongodb-2
              openshift.io/scc: restricted
Status:       Running
IP:           10.128.2.168
IPs:
  IP:           10.128.2.168
Controlled By:  ReplicationController/mongodb-2
Containers:
  mongodb:
    Container ID:   cri-o://a8d475df4c5e6e8aadb9021d938d8520aa123bd91f92e43d4ede9b7631a749ac
    Image:          image-registry.openshift-image-registry.svc:5000/openshift/mongodb@sha256:ad5dc22e6115adc0d875f6d2eb44b2ba594d07330a600e67bf3de49e02cab5b0
    Image ID:       image-registry.openshift-image-registry.svc:5000/openshift/mongodb@sha256:ad5dc22e6115adc0d875f6d2eb44b2ba594d07330a600e67bf3de49e02cab5b0
    Port:           27017/TCP
    Host Port:      0/TCP
    State:          Running
      Started:      Fri, 14 Feb 2020 17:20:53 +0100
    Ready:          True
    Restart Count:  0
    Limits:
      memory:  512Mi
    Requests:
      memory:   512Mi
    Liveness:   tcp-socket :27017 delay=30s timeout=1s period=10s #success=1 #failure=3
    Readiness:  exec [/bin/sh -i -c mongo 127.0.0.1:27017/$MONGODB_DATABASE -u $MONGODB_USER -p $MONGODB_PASSWORD --eval="quit()"] delay=3s timeout=1s period=10s #success=1 #failure=3
    Environment:
      MONGODB_USER:            <set to the key 'database-user' in secret 'mongodb'>            Optional: false
      MONGODB_PASSWORD:        <set to the key 'database-password' in secret 'mongodb'>        Optional: false
      MONGODB_ADMIN_PASSWORD:  <set to the key 'database-admin-password' in secret 'mongodb'>  Optional: false
      MONGODB_DATABASE:        <set to the key 'database-name' in secret 'mongodb'>            Optional: false
    Mounts:
      /newdata1 from newdata1 (rw)
      /var/lib/mongodb/data from mongodb-data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-nz6m4 (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  mongodb-data:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  mongodb
    ReadOnly:   false
  newdata1:
    Type:       PersistentVolumeClaim (a reference to a PersistentVolumeClaim in the same namespace)
    ClaimName:  newdata1
    ReadOnly:   false
  default-token-nz6m4:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-nz6m4
    Optional:    false
QoS Class:       Burstable
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/memory-pressure:NoSchedule
                 node.kubernetes.io/not-ready:NoExecute for 300s
                 node.kubernetes.io/unreachable:NoExecute for 300s
Events:
  Type     Reason                  Age        From                              Message
  ----     ------                  ----       ----                              -------
  Warning  FailedScheduling        <unknown>  default-scheduler                 pod has unbound immediate PersistentVolumeClaims (repeated 2 times)
  Warning  FailedScheduling        <unknown>  default-scheduler                 pod has unbound immediate PersistentVolumeClaims (repeated 2 times)
  Normal   Scheduled               <unknown>  default-scheduler                 Successfully assigned przemek/mongodb-2-plpgl to worker02.ocp4.scale.com
  Normal   SuccessfulAttachVolume  14m        attachdetach-controller           AttachVolume.Attach succeeded for volume "pvc-b8989e44-8a81-469f-9ec9-b2f9f3345d0a"
  Normal   SuccessfulAttachVolume  14m        attachdetach-controller           AttachVolume.Attach succeeded for volume "pvc-de5b1ecd-2ceb-4456-8287-ebcbed37fc76"
  Normal   Pulled                  14m        kubelet, worker02.ocp4.scale.com  Container image "image-registry.openshift-image-registry.svc:5000/openshift/mongodb@sha256:ad5dc22e6115adc0d875f6d2eb44b2ba594d07330a600e67bf3de49e02cab5b0" already present on machine
  Normal   Created                 14m        kubelet, worker02.ocp4.scale.com  Created container mongodb
  Normal   Started                 14m        kubelet, worker02.ocp4.scale.com  Started container mongodb
```

```
[root@worker01 ~]# oc get pvc
NAME            STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS                      AGE
mongodb         Bound    pvc-b8989e44-8a81-469f-9ec9-b2f9f3345d0a   1Gi        RWO            ibm-spectrum-scale-csi-remotefs   18m
newdata1        Bound    pvc-de5b1ecd-2ceb-4456-8287-ebcbed37fc76   1Gi        RWX            ibm-spectrum-scale-csi-remotefs   15m
postgres-data   Bound    pvc-21490825-c69f-4bef-bf97-1d09dec2ede7   1Gi        RWO            ibm-spectrum-scale-csi-remotefs   5h38m
```

With df -hT I cannot see the first mount point `/var/lib/mongodb/data`.
```
[root@worker01 ~]# oc rsh mongodb-2-plpgl
sh-4.2$ df -hT
Filesystem             Type     Size  Used Avail Use% Mounted on
overlay                overlay  200G   44G  157G  22% /
tmpfs                  tmpfs     64M     0   64M   0% /dev
tmpfs                  tmpfs     63G     0   63G   0% /sys/fs/cgroup
shm                    tmpfs     64M     0   64M   0% /dev/shm
tmpfs                  tmpfs     63G  4.1G   59G   7% /etc/passwd
remote_fs0_1m          gpfs      15T   61G   15T   1% /newdata1
/dev/mapper/sr650-root xfs      200G   44G  157G  22% /etc/hosts
tmpfs                  tmpfs     63G   24K   63G   1% /run/secrets/kubernetes.io/serviceaccount
tmpfs                  tmpfs     63G     0   63G   0% /proc/acpi
tmpfs                  tmpfs     63G     0   63G   0% /proc/scsi
tmpfs                  tmpfs     63G     0   63G   0% /sys/firmware
```

But the two mount points can be seen in `mount` output.
```
sh-4.2$ mount |grep gpfs
remote_fs0_1m on /newdata1 type gpfs (rw,relatime,seclabel)
remote_fs0_1m on /var/lib/mongodb/data type gpfs (rw,relatime,seclabel)
```

Can you fix the issue so that I can see the two mount points in `df` output?

**Comment by deeghuge** [comment]
I did some experiments around this. During experiments I saw df is showing only one entry per gpfs filesystem in the df listing. if there are two or more PVC from same gpfs filesystem then only one entry will be listed in df output. If there are 2 pvc from 2 different filesystem then two entries will be listed in df output. 

CSI Driver has no role in the df listing so either this can be moved to core scale for further investigation or cancelled as WAD.
